package/kedro_viz/integrations/kedro/data_loader.py

```diff
@@ -114,3 +120,36 @@ def load_data(
 
     context = load_context(project_path=project_path, env=env)
     return context.catalog, context.pipelines
+
+
+def load_data_for_all_versions(
+    filepath: str, limit: int = 10
+) -> Optional[Dict[datetime, Any]]:
+    """Load data for all versions of the dataset
+    Args:
+        filepath: the path where the dataset is located.
+        limit: the maximum number of past versions we want to load.
+    Returns:
+        A dictionary containing the version and the json data inside each version
+    """
+    version_list = [
+        path for path in sorted(Path(filepath).iterdir(), reverse=True) if path.is_dir()
+    ]
+    versions = {}
+    for index, version in enumerate(version_list):
```

Comment:
You can simplify this a bit by doing:
```suggestion
    for version in version_list[:limit]
```

And then you can remove the `if index == limit` below.